db/db_impl/db_impl.cc

```diff
@@ -851,6 +857,37 @@ Status DBImpl::RegisterRecordSeqnoTimeWorker() {
   if (seqno_time_cadence == 0) {
     s = periodic_task_scheduler_.Unregister(PeriodicTaskType::kRecordSeqnoTime);
   } else {
+    if (GetLatestSequenceNumber() == 0) {
+      // New DB. First, we don't want to leave latest seqno as 0 so that we
+      // can properly check in RecordSeqnoToTimeMapping that we aren't
+      // recording bad records (e.g. in a read-only DB). Second, to support
+      // data import with an older write time, we pre-allocate some sequence
+      // numbers pre-mapped over the time period we care about. But we can
+      // only do the second one predictably if called from DB::Open(),
+      // otherwise there could be a race between CreateColumnFamily and the
+      // first Write to the DB that determines whether the sequence numbers
+      // are pre-allocated.
+      assert(seqno_to_time_mapping_.Empty());
+
+      if (from_db_open) {
+        // We can simply modify these, before writes are allowed
+        constexpr uint64_t kMax = SeqnoToTimeMapping::kMaxSeqnoTimePairsPerSST;
```

Comment:
What about using `seqno_to_time_mapping_.GetCapacity()` as the number of sequence to reserve here? It will make the pre-populated window's pairs have the same cadence as the to be populated future.